src/main/java/seedu/address/model/application/Application.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model.application;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import seedu.address.model.tag.Tag;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents an Application in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Application {
+
+    // Identity fields
+    private final CompanyName companyName;
+    private final ContactNumber contactNumber;
+    private final Email email;
+    private final PositionApplied positionApplied;
+
+    // Data fields
+    private final Set<Tag> tags = new HashSet<>();
```

Comment:
I think our plan for v1.2 doesn't have tags. Only company name, date, and position. I can add these in Parser too but we might have to edit User Guide?

Date is missing too but you can probably use LocalDate for it.